sync.py

```diff
@@ -32,348 +26,187 @@
-        if 'seminary' in block_properties['path']:
-            return my_path + " has 'seminary' in its path"
+                # we use a break since we delete the amm down below
+                break
+            counter += 1
 
-        nodes = block_properties['structuredData']['structuredDataNodes']['structuredDataNode']
+        # delete all after the one you exited at
+        del find(concentration, 'concentration_banner', False)[counter + 1:]
 
-        # mark the code down as "seen"
-        try:
-            program_hash = nodes[0]['structuredDataNodes']['structuredDataNode'][0]['text']
-            self.hashes.add(program_hash)
-        except KeyError:
-            # not all programs have generic codes -- only concentration codes.
-            pass
+        return True
 
-        # every node after the first is a concentration
-        for concentration_structure in nodes[1:]:
+    def process_block(self, block_id):
+        program_block = Block(self.cascade, block_id)
+        block_asset = program_block.asset
 
-            concentration = concentration_structure['structuredDataNodes']['structuredDataNode']
+        block_path = find(block_asset, 'path', False)
+        if find(block_asset, 'definitionPath', False) != "Blocks/Program":
+            return block_path + " not in Blocks/Program"
 
-            try:
-                concentration_code = concentration[0]['text']
+        # gather concentrations
+        concentrations = find(program_block.structured_data, 'concentration')
+        if not isinstance(concentrations, list):
+            concentrations = [concentrations]
 
-                # load the data from banner for this code
-                data = self.get_data_for_code(concentration_code)
-                if len(data) > 0:
-                    concentration_code_has_data = True
-                else:
-                    concentration_code_has_data = False
+        for concentration in concentrations:
+            concentration_code = find(concentration, 'concentration_code', False)
 
-                # If the concentration code ('2-MA-COUG' for example) is in the list of programs to skip,
-                # continue iterating over the concentrations in this block
-                if concentration_code in SKIP_CODES:
-                    print "Code '%s' found in skip list; skipping it" % concentration_code
-                    if concentration_code_has_data:
-                        print "Although code '%s' is being skipped, it has data in Banner." % concentration_code
-                    print ""
+            if concentration_code in SKIP_CODES:
+                print "Code '%s' found in skip list; skipping it" % concentration_code
+                continue
 
+            self.delete_and_clear_cohort_details(concentration)
+
+            # todo: remove this after launch
+            ##################### Code to be used until after we launch! ##########################
```

Comment:
Is this comment still true? If this PR gets merged, wouldn't this be "launched"?

Reply:
The comment of `todo: remove this after launch` is still true. The reason this isn't removed yet, is it is used for testing. In my PR description, I mention this is not ready to be merged

Reply:
👍